cosmwasm/enclaves/shared/block-verifier/src/verify/registration.rs

```diff
@@ -0,0 +1,43 @@
+use crate::VERIFIED_MESSAGES;
+use log::{debug, error};
+use protobuf::Message;
+
+pub fn verify_reg_msg(certificate: &[u8]) -> bool {
+    let mut verified_msgs = VERIFIED_MESSAGES.lock().unwrap();
```

Comment:
Not relevant specifically to this PR but just a thought, should we log a certain error in case this `lock().unwrap()` fails? Debugging this would be hell

Reply:
I actually think it would be kind of dramatic, as this would mean a failure in the lock or something? Not sure when this is even possible

Reply:
Yeah, it's going to give no details if it fails, but it should never fail.